omegaconf/basecontainer.py

```diff
@@ -103,7 +103,19 @@ def __setstate__(self, d: Dict[str, Any]) -> None:
 
```

Comment:
I wanted to mention a pattern I've seen in the past: including a version number in the dict returned from `__getstate__`, and looking for the version number in `__setstate__`.

So for example, in `__getstate__`, something like `dict_copy["_omegaconf_version"] = omegaconf.__version__`.

Reply:
Though I guess such a heavy-duty technique is not necessary in this case...

Reply:
If we ever want to make a big change to the serialization format we can consider adding such a version.
It seems a bit too heavy for this use case. Also, we are saving many objects, so that version will be replicated a zillion times.

Reply:
Could be added to the root only. But I agree this can probably wait until the next major change to serialization.